### DISCUSSION TITLE
Edge of Node metrics

### DISCUSSION THREAD

**Discussion by mmoraes-weir** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#19] [Ideas]
Hello Thomas.

First of all, we all appreciate your effort on this project.

I can't tell if this is a feature the community would like to see, but in the sense of making it better and better, is in your pipeline an implementation of Node metrics? Let's say the Raspberry (managing several devices) metrics, like temperature and other information?

Cheers.

**Comment by thomassorensen2650** [comment]
Hi 

I like the idea, and it's definitely something we should get implemented at some point. Only question is what to do about the values on NBIRTH as NBIRTH is send when node-red is initialized, and before user flows are executed -  but I guess that we can just set the metrics to NULL on birth and then users set them right away.

**Reply by mmoraes-weir** [reply]
Hi.

I think one other is to send the NBIRTH it already sends, with 'Node Control/Rebirth' and 'bdSeq', with its correspondent values, and, when the new/extra metrics are defined in the node, it republishes the "standard" metric, with the current values, with the newly added metrics. As for these new metrics, if no value is defined, then send NULL, as it best represent the unknown condition of that metric.

Just a thought. I think we could build on the idea of metrics.
